test/e2e/resourceinterpreter_test.go

```diff
@@ -422,4 +422,208 @@ end`,
+				"interpreter-customization"+rand.String(RandomStrLength),
+				configv1alpha1.CustomizationTarget{
+					APIVersion: "apps/v1",
+					Kind:       "Deployment",
+				},
+				configv1alpha1.CustomizationRules{
+					Retention: &configv1alpha1.LocalValueRetention{
+						LuaScript: `
+function Retain(desiredObj, observedObj)
+  desiredObj.spec.paused = observedObj.spec.paused
+  return desiredObj
+end`,
+					},
+				})
+		})
+
+		ginkgo.It("Retain testing", func() {
+			ginkgo.By("update workload's spec.paused to true", func() {
+				clusterClient := framework.GetClusterClient(targetCluster)
+				gomega.Expect(clusterClient).ShouldNot(gomega.BeNil())
+				var memberDeploy *appsv1.Deployment
+				framework.WaitDeploymentPresentOnClusterFitWith(targetCluster, deployment.Namespace, deployment.Name,
+					func(deployment *appsv1.Deployment) bool {
+						memberDeploy = deployment
+						return true
+					})
+				memberDeploy.Spec.Paused = updatedPaused
+				framework.UpdateDeployment(clusterClient, memberDeploy)
+			})
+
+			// Wait executeController to reconcile then check if it is retained
+			time.Sleep(waitTime)
+			ginkgo.By("check if workload's spec.paused is retained", func() {
+				gomega.Eventually(func(g gomega.Gomega) (bool, error) {
+					var memberDeployment *appsv1.Deployment
+					framework.WaitDeploymentPresentOnClusterFitWith(targetCluster, deployment.Namespace, deployment.Name,
+						func(deployment *appsv1.Deployment) bool {
+							memberDeployment = deployment
+							return true
+						})
+					return memberDeployment.Spec.Paused, nil
+				}, pollTimeout, pollInterval).Should(gomega.Equal(updatedPaused))
+			})
+		})
+	})
+
+	ginkgo.Context("InterpreterOperation AggregateStatus testing", func() {
+		ginkgo.BeforeEach(func() {
+			customization = testhelper.NewResourceInterpreterCustomization(
+				"interpreter-customization"+rand.String(RandomStrLength),
+				configv1alpha1.CustomizationTarget{
+					APIVersion: "apps/v1",
+					Kind:       "Deployment",
+				},
+				configv1alpha1.CustomizationRules{
+					StatusAggregation: &configv1alpha1.StatusAggregation{
+						LuaScript: `
+function AggregateStatus(desiredObj, statusItems)
+	if statusItems == nil then
+		return desiredObj
+	end
+	if desiredObj.status == nil then
+		desiredObj.status = {}
+	end
+	replicas = 0
+	for i = 1, #statusItems do
+		if statusItems[i].status ~= nil and statusItems[i].status.replicas ~= nil  then
+			replicas = replicas + statusItems[i].status.replicas + 1
+		end 
+	end
+	desiredObj.status.replicas = replicas
+	return desiredObj
+end`,
+					},
+				})
+		})
+		ginkgo.It("AggregateStatus testing", func() {
+			framework.WaitDeploymentPresentOnClusterFitWith(targetCluster, deployment.Namespace, deployment.Name,
+				func(deployment *appsv1.Deployment) bool {
+					return true
+				})
+			ginkgo.By("check whether the deployment status can be correctly collected", func() {
+				wantedReplicas := *deployment.Spec.Replicas + 1
```

Comment:
Here can add some comments. such as:
// Only in the current case, a special example is constructed to distinguish the build-in logic.